src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -133,9 +167,13 @@ export class StickyLineCandidateProvider extends Disposable {
 
 class StickyOutlineElement {
 	public static fromOutlineModel(outlineModel: OutlineModel | OutlineElement | OutlineGroup): StickyOutlineElement {
-		const children = [...outlineModel.children.values()].map(child =>
-			StickyOutlineElement.fromOutlineModel(child)
-		);
+		const children = [...outlineModel.children.values()].map(child => {
+			if (child instanceof OutlineElement && child.symbol.selectionRange.startLineNumber !== child.symbol.range.endLineNumber || child instanceof OutlineGroup || child instanceof OutlineModel) {
+				return StickyOutlineElement.fromOutlineModel(child);
+			} else {
+				return;
+			}
+		}).filter((child) => !!child) as StickyOutlineElement[];
```

Comment:
Looking at this with the performance-eye: 

* the `[...something.values()]`-statement iterates over all values and allocates a new array
* the `map`-call iterates again over all values and also creates a new array
* the `filter`-call iterates again and creates yet another array

It's better to iterate just once using a forof-loop and some if-statement

Reply:
That's right, thanks for pointing it out